byway/src/main/webapp/utils.js

```diff
@@ -0,0 +1,33 @@
+/** Script that contains shared functions and variables */
+
+// map object used in the route page
+let map; 
+
+/**
+ * Uses placeId to retrieve details like coordinates, place name, etc.
+ * @param {String} placeId a textual identifier that uniquely identifies a place
+ * @return {Promise} result a Place object
+ */
+function findPlace(placeId) {
+  const request = {
+    placeId: placeId,
+    fields: ['name', 'geometry', 'place_id', 'photos', 'formatted_address']
+  }
+  const result = new Promise((resolve, reject) => {
+    placesService.getDetails(request, (result, status) => {
+      if(status == "OK") {
+        resolve(result);
+      } else {
+        alert("Status: " + status);
+        reject(new Error("Could not retrieve place result object from request."));
+      }
+    })
+  }).catch(error => {
+      alert(error);
+  });
+
+  return result;
+}
+
+/* exported findPlace, map, placesService */
+/* global placesService:true */
```

Comment:
I don't think you need the `:true` here. Where did you find this `name:true` syntax?

Reply:
I said it in the other PR, I needed to add this because I got an error saying global vars can't be modified.